### DISCUSSION TITLE
Next.js prebuilt deploy error ENOENT: no such file or directory, stat '/vercel/path0/.vercel/output/functions/[locale]/appointments/[id].func'

### DISCUSSION THREAD

**Discussion by mbpictures** [vercel/community#5918] [Help]
### Summary

Hi!

I have a next.js project inside a monorepo with dependencies to other local packages. Therefore I tried to prebuilt the project locally and upload it (as described in the docs) using this command:
```vercel build && vercel deploy --prebuilt```

However, when I run this command I get this error:
```
Error: ENOENT: no such file or directory, stat '/vercel/path0/.vercel/output/functions/[locale]/appointments/[id].func'
```

I'm using the new App Router of Next.js together with next-intl localization (which works fine in development). My folder structure is something like this:
```
- root
  - packages
    - ...
    - next
      - src
        - middleware.tsx 
        - app
          - [locale]
            - page.tsx
            - layout.tsx
            - appointments
              - page.tsx 
              - [id]
                - page.tsx
              
```

When I check the ```.vercel/output``` folder, I can see the files for all routes I defined, also the one that doesn't seem to exist on vercel servers.

### Steps to Reproduce

- Create a Next.js project
- Add dynamic routes to the app router
- Add middleware to redirect to correct locale
- Deploy with local prebuilt: ```vercel build && vercel deploy --prebuilt```

**Comment by mbpictures** [comment]
Running the deploy command on WSL, I get another error
```
Error: ENOENT: no such file or directory, lstat '/node_modules/styled-jsx/index.js'
```
Even setting ```VERCEL_FORCE_NO_BUILD_CACHE``` to 1 doesn't fix this error.

EDIT:
Downgrading to vercel cli 33.2.0 fixes the error during deploy, but I'm getting this server error when trying to access the pages:
```
Cannot find module './node-environment'
Require stack:
- /var/task/node_modules/next/dist/server/next-server.js
- /var/task/___next_launcher.cjs
Did you forget to add it to "dependencies" in `package.json`?
Child exited with error exit status 1INIT_REPORT Init Duration: 293.68 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
Cannot find module './node-environment'
Require stack:
- /var/task/node_modules/next/dist/server/next-server.js
- /var/task/___next_launcher.cjs
Did you forget to add it to "dependencies" in `package.json`?
INIT_REPORT Init Duration: 372.10 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Error: Runtime exited without providing a reason
```

**Comment by amyegan** [comment]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) we can use to help with debugging?

**Reply by mbpictures** [reply]
Thanks for your fast reply! I created [this](https://github.com/mbpictures/vercel-prebuilt-deploy-bug-reproduction) reproducible example, hope it helps in the investigation. However, I encountered some different but similar error in the server logs:
```
TypeError: Cannot read properties of undefined (reading 'clientModules')
    at /var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:199008
    at rW (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:16:14861)
    at nM (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:47419)
    at nN (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64674)
    at nM (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61233)
    at nN (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64674)
    at nI (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:46806)
    at nM (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:47570)
    at nN (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64674)
    at nM (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61233)
```

**Reply by mbpictures** [reply]
Following https://github.com/vercel/next.js/issues/53728#issuecomment-1719673747 this tipps didn't work either. Also, an interesting fact I just found out: during deployment (--prebuilt), only 500KB are uploaded while the .vercel/output folder has a size of 7MB.

**Reply by mbpictures** [reply]
Any chance for a fix or workaround as I want to go live soon?
